[Common] [TOF] streamline ccdb configuration

njacazio opened the pull request.

alibuild commented:
Error while checking build/O2Physics/o2 for 123a25fdb58db868ee718f46d6a781adde766c84 at 2024-11-28 07:28:
```
## sw/BUILD/O2Physics-latest/log
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/Common/TableProducer/PID/pidTOFbeta.cxx:127:3: error: 'o2::pid::tof::Beta' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/Common/TableProducer/PID/pidTOFbeta.cxx:145:27: error: 'o2::pid::tof::TOFMass' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/Common/TableProducer/PID/pidTOFbeta.cxx:147:27: error: 'o2::pid::tof::TOFMass' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/Common/TableProducer/PID/pidTOFMerge.cxx:1488:3: error: 'o2::pid::tof::Beta' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/Common/TableProducer/PID/pidTOFMerge.cxx:1512:3: error: 'o2::pid::tof::Beta' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/Common/TableProducer/PID/pidTOFMerge.cxx:1503:27: error: 'o2::pid::tof::TOFMass' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/Common/TableProducer/PID/pidTOFMerge.cxx:1505:27: error: 'o2::pid::tof::TOFMass' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/Common/TableProducer/PID/pidTOFMerge.cxx:1528:27: error: 'o2::pid::tof::TOFMass' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/Common/TableProducer/PID/pidTOFMerge.cxx:1530:27: error: 'o2::pid::tof::TOFMass' is not a template
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/8692/123a25fdb58db868ee718f46d6a781adde766c84/build_O2Physics_o2/pretty.html).

alibuild commented:
Error while checking build/O2Physics/o2 for 1448d1143106987e58cc3b764db6eccb1935a3bd at 2024-11-28 07:39:
```
## sw/BUILD/O2Physics-latest/log
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/DPG/Tasks/AOTTrack/PID/TOF/qaPIDTOFEvTime.cxx:338:58: error: 'o2::pid::tof::Beta' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/DPG/Tasks/AOTTrack/PID/TOF/qaPIDTOFEvTime.cxx:339:64: error: 'o2::pid::tof::Beta' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/DPG/Tasks/AOTTrack/PID/TOF/qaPIDTOFEvTime.cxx:340:64: error: 'o2::pid::tof::Beta' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/DPG/Tasks/AOTTrack/PID/TOF/qaPIDTOFEvTime.cxx:341:57: error: 'o2::pid::tof::Beta' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/DPG/Tasks/AOTTrack/PID/TOF/qaPIDTOFEvTime.cxx:343:58: error: 'o2::pid::tof::TOFMass' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/DPG/Tasks/AOTTrack/PID/TOF/qaPIDTOFEvTime.cxx:344:64: error: 'o2::pid::tof::TOFMass' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/DPG/Tasks/AOTTrack/PID/TOF/qaPIDTOFEvTime.cxx:345:64: error: 'o2::pid::tof::TOFMass' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/DPG/Tasks/AOTTrack/PID/TOF/qaPIDTOFEvTime.cxx:346:57: error: 'o2::pid::tof::TOFMass' is not a template
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/8692/1448d1143106987e58cc3b764db6eccb1935a3bd/build_O2Physics_o2/pretty.html).

alibuild commented:
Error while checking build/O2Physics/o2 for 64a70a251cc51e0f9239e0a6497ca4be6ecf16ac at 2024-11-28 08:06:
```
## sw/BUILD/O2Physics-latest/log
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/PWGLF/TableProducer/Nuspex/LFTreeCreatorClusterStudies.cxx:268:3: error: 'o2::pid::tof::Beta' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/PWGLF/TableProducer/Nuspex/LFTreeCreatorClusterStudies.cxx:269:3: error: 'o2::pid::tof::Beta' is not a template
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/8692/64a70a251cc51e0f9239e0a6497ca4be6ecf16ac/build_O2Physics_o2/pretty.html).

alibuild commented:
Error while checking build/O2Physics/o2 for 915624afe65eedee81f6756425b79ec0629d4009 at 2024-11-28 08:28:
```
## sw/BUILD/O2Physics-latest/log
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/PWGLF/TableProducer/Nuspex/hyperRecoTask.cxx:158:3: error: 'o2::pid::tof::Beta' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/PWGLF/TableProducer/Nuspex/nucleiSpectra.cxx:498:5: error: 'o2::pid::tof::Beta' is not a template
/sw/SOURCES/O2Physics/8692-slc9_x86-64/0/PWGLF/TableProducer/Nuspex/he3HadronFemto.cxx:595:5: error: 'o2::pid::tof::Beta' is not a template
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/8692/915624afe65eedee81f6756425b79ec0629d4009/build_O2Physics_o2/pretty.html).

njacazio commented:
@fmazzasc I removed the template from the class and moved it to the static function for `GetBeta` this way it is easier to use, just double check the code pls.

njacazio commented:
This also catches the case where the run is changed in the same workflow and the pass from the metadata is taken and stored in the ccdb manager. Thanks to @rolavick

alibuild reviewed (approved):
Auto-approving on behalf of @njacazio.
